src/Tempest/Generation/src/DataObjects/StubFile.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Generation\DataObjects;
+
+use Exception;
+use Nette\InvalidStateException;
+use Tempest\Generation\ClassManipulator;
+use Tempest\Generation\Enums\StubFileType;
+use Tempest\Generation\Exceptions\ClassNotFoundException;
+use Tempest\Reflection\ClassReflector;
+use Throwable;
+
+/**
+ * Represents a file that is to be generated.
+ */
+final class StubFile
+{
+    public function __construct(
+        public readonly string $filePath,
+        public readonly StubFileType $type,
+    ) {
+    }
+
+    public static function fromFilePath(string $filePath): self
```

Comment:
Can't these two static constructors be combined into one? `::from($pathOrClass)`. You can check whether it's a path with `if(is_file($pathOrClass))`

It's just a minor detail, but makes the API a bit more convenient to use

Reply:
You're right, added 😄